builtin/logical/mongodb/path_config_lease.go

```diff
@@ -0,0 +1,92 @@
+package mongodb
+
+import (
+	"time"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfigLease(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/lease",
+		Fields: map[string]*framework.FieldSchema{
+			"ttl": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Description: "Default ttl for credentials.",
+			},
+
+			"max_ttl": &framework.FieldSchema{
+				Type:        framework.TypeDurationSecond,
+				Description: "Maximum time a set of credentials can be valid for.",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation:   b.pathConfigLeaseRead,
+			logical.UpdateOperation: b.pathConfigLeaseWrite,
+		},
+
+		HelpSynopsis:    pathConfigLeaseHelpSyn,
+		HelpDescription: pathConfigLeaseHelpDesc,
+	}
+}
+
+func (b *backend) pathConfigLeaseWrite(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	entry, err := logical.StorageEntryJSON("config/lease", &configLease{
+		TTL:    time.Second * time.Duration(d.Get("ttl").(int)),
+		MaxTTL: time.Second * time.Duration(d.Get("max_ttl").(int)),
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err := req.Storage.Put(entry); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (b *backend) pathConfigLeaseRead(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	leaseConfig, err := b.LeaseConfig(req.Storage)
+
+	if err != nil {
+		return nil, err
+	}
+	if leaseConfig == nil {
+		return nil, nil
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"ttl":     leaseConfig.TTL.String(),
```

Comment:
OK -- we've discussed internally. A lot of Vault APIs aren't consistent, but we are thinking towards a future `v2` API where things *are* consistent. In that case we want durations across the wire in JSON to be easily machine parseable, and format for the user in the last mile. So please use `Seconds()` rather than `String()` here.